src/DeviceHardDriveType.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+class DeviceHardDriveType extends CommonDeviceType
+{
+    public $additional_fields_for_dictionnary = ['manufacturer'];
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Device hard drive type', 'Device hard drive types', $nb);
+    }
+
+    /**
```

Comment:
I do not understand why it's required to use `getHtmlTable` here while it's not in other "DeviceXyzType`.

Reply:
It's require to display the Type column in Computer>Components page, like this : 
![image](https://github.com/glpi-project/glpi/assets/107540223/f4ca9f7d-e78e-41d7-8a6d-0ccd196276f3)

Without it, the columns disappear:
![image](https://github.com/glpi-project/glpi/assets/107540223/d1b3ecf9-2364-44a0-b831-1e237f7b983e)